moto/glue/models.py

```diff
@@ -254,62 +274,223 @@ def tag_resource(self, resource_arn, tags):
 
         if registry_name in self.registries:
             raise GSRAlreadyExistsException(
-                operation_name,
                 resource="Registry",
                 param_name="RegistryName",
                 param_value=registry_name,
             )
 
-        if description and len(description.encode("utf-8")) > max_description_length:
-            param_name = "description"
-            raise ResourceNameTooLongException(operation_name, param_name)
+        if description:
+            validate_description_pattern_and_length(description)
 
-        if description and re.match(registry_description_pattern, description) is None:
-            param_name = "description"
-            raise ParamValueContainsInvalidCharactersException(
-                operation_name, param_name
-            )
-
-        if tags and len(tags) > max_tags_allowed:
-            raise InvalidNumberOfTagsException(operation_name)
+        if tags and len(tags) > constants.MAX_TAGS_ALLOWED:
+            raise InvalidNumberOfTagsException()
 
         registry = FakeRegistry(registry_name, description, tags)
         self.registries[registry_name] = registry
-        return registry
+        return registry.as_dict()
+
+    def create_schema(
+        self,
+        registry_id,
+        schema_name,
+        data_format,
+        compatibility,
+        schema_definition,
+        description=None,
+        tags=None,
+    ):
+        if registry_id:
+            if registry_id.get("RegistryName") is not None:
+                if registry_id.get("RegistryArn") is not None:
+                    raise InvalidRegistryIdBothParamsProvidedException()
+                registry_name = registry_id.get("RegistryName")
+                validate_registry_name_pattern_and_length(registry_name)
+            else:
+                registry_arn = registry_id.get("RegistryArn")
+                validate_arn_pattern_and_length(registry_arn)
+                registry_name = registry_arn.split("/")[-1]
+        else:
+            registry_name = constants.DEFAULT_REGISTRY_NAME
+
+        if registry_name not in self.registries:
+            if registry_name == constants.DEFAULT_REGISTRY_NAME:
+                self.create_registry(registry_name, registry_name)
+            else:
+                if registry_id.get("RegistryName"):
+                    raise GSREntityNotFoundException(
+                        resource="Registry",
+                        param_name="RegistryName",
+                        param_value=registry_name,
+                    )
+                if registry_id.get("RegistryArn"):
+                    raise GSREntityNotFoundException(
+                        resource="Registry",
+                        param_name="RegistryArn",
+                        param_value=registry_arn,
+                    )
+
+        registry = self.registries[registry_name]
+
+        validate_schema_name_pattern_and_length(schema_name)
```

Comment:
You can also split this function into multiple methods like "validateRegistry", "validateSchema" and "createSchema" to make it more readable.

Reply:
I did not get you here, there are dedicated methods for validating registry_name, schema_name, description etc. Maybe the name is confusing. Please elaborate.